src/main/java/seedu/tp/flashcard/FlashcardList.java

```diff
@@ -81,11 +83,15 @@ public boolean equals(Object obj) {
             return false;
         }
 
-        for (int idx = 0; idx < this.getTotalFlashcardNum(); idx++) {
-            if (!this.getFlashcardAtIdx(idx).equals(otherFlashcardList.getFlashcardAtIdx(idx))) {
+        for (int idx = 0; idx < otherFlashcardList.getTotalFlashcardNum(); idx++) {
```

Comment:
If we want to do sorting, we need to do it based on something. Since the indexes of the flashcards are given following the order of additions, we can not sort the list with them.
In this way, I am not quite sure how to do sorting now. Maybe we can figure this out in further version.

Reply:
I implemented a way to sort Flashcards based on startDate to implement the timeline feature. Although I'm not really sure we need to worry about sorting the flashcards here. The main purpose of having the equals method for FlashcardList as of now (afaik) is so we can use assertEquals on FlashcardLists, and in the tests I can see it being useful that we can check the order that flashcards being inserted is the same in two FlashcardLists. If we need to have an unordered equality check in the future perhaps we can make another method for that.

Reply:
When we compare groups, I think we do need an unordered equality check for the flashcard lists of the groups. So I am not sure now we need to make this equal method ordered or unordered.

Reply:
I will update the test case to avoid groups with unordered equal flashcard lists. So maybe we can write methods for unordered equality check in the future.

Reply:
oh i see.
u can just do the sorting here if u want or can make another method. whichever is easier for u i guess. Here is a code snippet for how to get a sorted list of flashcards.
```
List<Flashcard> flashcards = new ArrayList<>(flashcardList.getFlashcards());
Collections.sort(flashcards);
```

Reply:
i didnt consider that u need unordered equality check to compare groups. that makes sense

Reply:
OK, I will try to do sorting here.